h/services/search_index/_queue.py

```diff
@@ -112,6 +112,33 @@ def add_annotations_between_times(self, start_time, end_time, tag, force=False):
         )
         mark_changed(self._db)
 
+    def add_users_annotations(self, userid, tag, force=False, schedule_in=None):
+        """
+        Queue all a user's annotations to be synced to Elasticsearch.
+
+        See Queue.add() for documentation of the params.
+
+        :param userid: The ID of the user in "acct:USERNAME@AUTHORITY" format
+        :type userid: unicode
+        """
+        self._db.execute(
+            Job.__table__.insert().from_select(
+                [Job.name, Job.scheduled_at, Job.priority, Job.tag, Job.kwargs],
+                select(
+                    [
+                        text("'sync_annotation'"),
+                        text(f"'{self._datetime_at(schedule_in)}'"),
+                        text("100"),
+                        text(repr(tag)),
+                        func.jsonb_build_object(
+                            "annotation_id", Annotation.id, "force", force
+                        ),
+                    ]
+                ).where(Annotation.userid == userid),
```

Comment:
I've just realised that I don't actually understand how this works. Does SQLAlchemy just automatically know which table to select from because we referenced `Annotation` in the where clause? 

If this part didn't have a where, presumably there would be nothing to tip it off?

Reply:
I've no idea how SQLA inteprets it internally. `models.Annotation` is mentioned both in the `select()` (`select([..., func.jsonb_build_object("annotation_id", Annotation.id, "force", force)])`) and then later on in the `where()`'s as well. I assume it gathers up all the tables mentioned in any of the `select()`'s or `where()`'s and generates a `from` including them all.